tools/Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM python:3.10-slim-bullseye
+MAINTAINER Opentrons Engineering <engineering@opentrons.com>
+
+ENV BUILDENV_ROOT=/build-environment
+ENV POETRY_HOME=/build-environment/poetry
+ENV POETRY_VERSION=1.2.1
+ENV PATH=$PATH:/build-environment/poetry/bin
+ENV POETRY_VIRTUALENVS_IN_PROJECT=true
+ENV HOME=/build-environment
+
+RUN apt update && apt install -y curl wget file git
+RUN rm /bin/sh && ln -s /bin/bash /bin/sh
+RUN mkdir /build-environment
+ADD . /build-environment/tools
```

Comment:
Are we really using all the files in here? For instance, the README will get added in, if anything changes in it, it's gonna break your cache.

Also, is there a reason you are using `ADD` over `COPY`? https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy

Generally unless you are pulling from a URL or unpacking tar files